<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>waveform_generator</class>
 <widget class="QWidget" name="waveform_generator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>279</width>
    <height>273</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>277</width>
     <height>269</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="main_layout" stretch="1,0">
    <item>
     <widget class="PlotWidget" name="display"/>
    </item>
    <item>
     <layout class="QHBoxLayout" name="controls_layout" stretch="1,1">
      <item>
       <layout class="QVBoxLayout" name="column_1">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <layout class="QVBoxLayout" name="waveform_layout">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLabel" name="waveform_label">
            <property name="text">
             <string>Waveform</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="waveform">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="editable">
             <bool>false</bool>
            </property>
            <item>
             <property name="text">
              <string>Triangle</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Square</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Sine</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Ramp</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>x^2</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>x^3</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>x^4</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="amplitude_layout">
          <item>
           <widget class="QLabel" name="amplitude_label">
            <property name="text">
             <string>Amplitude</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="FormattedDoubleSpinBox" name="amplitude">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="maximum">
             <double>9999999999999999583119736832.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="offset_layout">
          <item>
           <widget class="QLabel" name="offset_label">
            <property name="text">
             <string>Offset</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="FormattedDoubleSpinBox" name="offset">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="minimum">
             <double>-1000000000000000000000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>99999999999999991433150857216.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="column_2">
        <item>
         <layout class="QVBoxLayout" name="points_layout">
          <item>
           <widget class="QLabel" name="points_label_2">
            <property name="text">
             <string>Points</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="points">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>9999999</number>
            </property>
            <property name="value">
             <number>101</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="periods_layout">
          <item>
           <widget class="QLabel" name="periods_label">
            <property name="text">
             <string>Periods</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="FormattedDoubleSpinBox" name="periods">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="maximum">
             <double>100.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="phase_layout">
          <item>
           <widget class="QLabel" name="phase_label">
            <property name="text">
             <string>Phase</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="FormattedDoubleSpinBox" name="phase">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="minimum">
             <double>-360.000000000000000</double>
            </property>
            <property name="maximum">
             <double>360.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
  <customwidget>
   <class>FormattedDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>custom_widgets</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
